File: dotfiles/local_bin/yesno.pl
#!/usr/bin/env perl

use strict;
use warnings;

use Carp;
use IPC::Open2;

my $q = "really???";
$q = $ARGV[0] if scalar @ARGV >0;

my $pid = open2(
  my $stdout, my $stdin,
  "rofi -dmenu -p '$q' -sep '\\0' -eh 2 -markup-rows -format i"
 ) or confess;

print $stdin "<span size='x-large' weight='heavy'>Yes</span>\0";
print $stdin "<span size='x-large' weight='heavy'>No</span>\0";
close($stdin);

my $stdout_ = do { local($/); <$stdout> };
close($stdout);
chomp $stdout_;

waitpid($pid, 0);
my $exit_code = $? >> 8;
# print STDERR "exit_code: $exit_code\n";

if ($exit_code == 0 && $stdout_ == 0) {
  exit 0;
} else {
  exit -1;
}

File: dotfiles/local_bin/+archive+/yesno.php
#!/usr/bin/env php
<?php

$q = "really???";
if(count($argv) > 1){
    $q = $argv[1];
}

$cmd = "rofi -theme-str 'window {width: 200px; height: 150px;}' -dmenu -p '$q' -sep '\\0' -eh 2 -markup-rows -format i";
$proc = proc_open($cmd, [
   0 => ["pipe", "r"],
   1 => ["pipe", "w"],
   2 => ["pipe", "r"],
], $pipes);

if (is_resource($proc)) {
    fwrite($pipes[0], "<span size='x-large' weight='heavy'>Yes</span>\0");
    fwrite($pipes[0], "<span size='x-large' weight='heavy'>No</span>\0");
    fclose($pipes[0]);

    $stdout = stream_get_contents($pipes[1]);
    $stdout = trim($stdout);
    fclose($pipes[1]);

    $return_value = proc_close($proc);
    echo "command returned: $return_value\n";
    echo "stdout[$stdout]\n";

    if($return_value == 0 && $stdout === '0'){
        exit(0);
    }else{
        exit(-1);
    }
}
